### PR TITLE
Add new plugin, `kubectl resources`

### PR DESCRIPTION
**Pull request by howardjohn**
Fixes https://github.com/howardjohn/kubectl-resources/issues/24

### CODE REVIEW

**Comment by ahmetb** [comment]
Thanks for submitting. Do you mind explaining how is it different than the several resource/allocation plugins we have?

**Comment by ahmetb** [comment]
I guess it is similar to view-allocations, just curious if it’s entirely overlapping or not.

**Comment by howardjohn** [comment]
As far as I know this is the only one aggregating resource requests/limits with usage. I realize it isn't groundbreakingly innovative, I made it since I find it useful and someone opened an issue recommending me to add it to krew

**Comment by ahmetb** [comment]
For reference, I'm attaching screenshots:

- kubectl view-allocations: 
image

- kubectl resources (with "pods" section of the "view-allocations" in the same screen) 
image

- `kubectl view-allocations` is also very similar to this, but it shows Node statuses.

The `kubectl view-allocations` has a funny separation of "cpu" "memory" and "pods", whereas `kubectl resources` is a cleaner representation of Pod resources.

Per our guidelines, this plugin likely would require renaming to accurately represent what it does (like `pod-resources`), or `pod-usage`. I'll have @corneliusweig also review and provide input.

**Comment by howardjohn** [comment]
You can also aggregate by namespace or node for what its worth, with `--by namespace` or `--by node`, so `pod-resources` doesn't seem quite accurate

**Comment by ahmetb** [comment]
Yeah, in that case --by=node part is overlapping with `resource-capacity` plugin (also overlapping in name). I believe @corneliusweig opened the issue so we are not leaving any plugin authors uninformed about Krew, but let's see if it fits. Right now there's considerable overlap with existing plugins.

image

File: plugins/resources.yaml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+          os: linux
+          arch: amd64
+      uri: https://github.com/howardjohn/kubectl-resources/releases/download/v0.1.0/kubectl-resources_0.1.0_Linux_x86_64.tar.gz
+      sha256: ad3d1a126b106a38fb92ae76bb728b00ed8b4ef02eef574456b73f21c28affb3
+      bin: kubectl-resources
+  shortDescription: >-
+    Access Kubernetes resource requests, limits, and usage.
+  homepage: https://github.com/howardjohn/kubectl-resources
+  description: |
+    Access Kubernetes resource requests, limits, and usage.
```
Comment (corneliusweig) [inline plugins/resources.yaml]: Maybe just briefly mention why `kubectl top` doesn't cut it?

**Comment by howardjohn** [comment]
I don't have any current plans to add any more features to the plugin (maybe a --watch flag, but not a high priority), but I am open to suggestions.

I am fine with changing the name, `resource-overview` sounds right. Let me know if that is sufficient or if I would need additional distinguishing

**Comment by ahmetb** [comment]
I'm good with both `resource-overview` and `resource-usage` (consider renaming your repo, and --help messages). 

This would help it distinguish from `resource-capacity` which is geared towards showing cluster-wide total and percentages, despite both plugins are largely overlapping.

**Comment by corneliusweig** [comment]
Then let's add this plugin to the index too. Please let us know if you plan to rename your repo as suggested.

File: plugins/resources.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+          os: linux
+          arch: amd64
+      uri: https://github.com/howardjohn/kubectl-resources/releases/download/v0.1.0/kubectl-resources_0.1.0_Linux_x86_64.tar.gz
+      sha256: ad3d1a126b106a38fb92ae76bb728b00ed8b4ef02eef574456b73f21c28affb3
+      bin: kubectl-resources
+  shortDescription: >-
+    Overview of Kubernetes resource requests, limits, and usage.
+  homepage: https://github.com/howardjohn/kubectl-resources
+  description: |
+    Overview of Kubernetes resource requests, limits, and usage. This plugin aggregates resource usage, similar to `kubectl top`,
```
Comment (corneliusweig) [inline plugins/resources.yaml]: Can you wrap this at ~80 chars for easier reading on small terminals?

**Comment by howardjohn** [comment]
Sounds good. I'll rename everything in my repo then send out an updated PR later. Thanks!

**Comment by saiharshitachava** [comment]
@howardjohn is this availble with krew yet?

**Pull request closed without merging**